pkg/scheduler/reconciliation/persistent.go

```diff
@@ -447,3 +447,135 @@ func (r *PersistentReconciliationRepository) UpdateOperationRetryID(schedulingID
 	}
 	return db.Transaction(r.Conn, dbOps, r.Logger)
 }
+
+func (r *PersistentReconciliationRepository) UpdateOperationPickedUp(schedulingID, correlationID string) error {
+	dbOps := func(tx *db.TxConnection) error {
+		op, err := r.GetOperation(schedulingID, correlationID)
+		if err != nil {
+			if repository.IsNotFoundError(err) {
+				r.Logger.Warnf("ReconRepo could not find operation (schedulingID:%s/correlationID:%s)", schedulingID, correlationID)
+			}
+			return err
+		}
+
+		if !op.PickedUp.IsZero() {
+			r.Logger.Warnf("Operation with schedulingID %s has already picked up at %s", schedulingID, op.PickedUp)
```

Comment:
A clearer message would be`Operation with schedulingID %s (was or has been) already picked up at %s`
Also, shouldn't we return an error when the `pickedUp` time is different from zero, because that means it was already updated, right?

Reply:
It is not an error since this could happen, when the operation was an orphan and is picked up again. but indeed there is a small bug, since to update the timestamp should only happen when an operation gets set to 'inProgress', and if it has already been picmed up before it should not warn the user but it should not update it again. Will fix it